### DISCUSSION TITLE
Is it possible to mute or remove a track?

### DISCUSSION THREAD

**Discussion by ar06x** [melanchall/drywetmidi#128] [Q&A] [answered]
I've searched the docs and haven't found anything on it. I'm currently using a `<List>` to keep track of muted tracks/channels

**Comment by melanchall** [comment, marked as answer]
Do you mean mute during playback? If so, you can create custom playback and ignore events on specified track chunk or channel. Please see example here: #122. So your implementation of `TryPlayEvent` will be like that:

```csharp
protected override bool TryPlayEvent(MidiEvent midiEvent, object metadata)
{
    if (mutedTrackChunks.Contains((int)metadata))
        return false;

    OutputDevice?.SendEvent(midiEvent);
    return true;
}
```

You need install [5.2.1-prerelease7](https://www.nuget.org/packages/Melanchall.DryWetMidi/5.2.1-prerelease7) version or later since provided API is not released yet.

**Reply by ar06x** [reply]
Currently I do: 
```
List<int> IgnoreTracks = new List<int>();
```
Then inside the playback `NoteCallback` I do this:
```
if (IgnoreTracks.Contains(rawNoteData.Channel)) return null;
```
Would your method be more efficient? Or which would be faster? That's what I'm currently trying to decide on.

**Reply by melanchall** [reply]
In your question you've mentioned **tracks**. If you need to mute MIDI events by track chunk index, you should use my method. But if you just need to mute **channels**, then your approach is the best way, so stick with it.
